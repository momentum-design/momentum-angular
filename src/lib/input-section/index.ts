export * from './input-section.component';
export * from './input-section.module';
