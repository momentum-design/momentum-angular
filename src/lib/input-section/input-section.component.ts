import {
  Component,
  HostBinding,
  Input,
} from '@angular/core';

export type InputSectionType = 'before' | 'after';

@Component({
  selector: 'md-input-section',
  template: `
    <ng-content></ng-content>
  `,
  styles: []
})

export class InputSectionComponent {
  /** @prop Determine the InputSection's position | 'before' */
  @Input() public position: InputSectionType = 'before';
  /** @prop Optional css class name | '' */
  @Input() public className: string;

  @HostBinding('class') get _class(): string {
    return (
      `md-input__${this.position || 'before'}` +
      `${this.className ? ` ${this.className}` : ''}`
    );
  }
}
