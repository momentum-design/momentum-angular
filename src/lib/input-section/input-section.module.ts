import { CommonModule } from '@angular/common';
import { NgModule, ModuleWithProviders } from '@angular/core';
import { InputSectionComponent } from './input-section.component';

@NgModule({
  declarations: [InputSectionComponent],
  imports: [CommonModule],
  exports: [InputSectionComponent],
})

export class InputSectionModule {
  static forRoot(): ModuleWithProviders {
    return {
      ngModule: InputSectionModule,
      providers: [],
    };
  }
}
