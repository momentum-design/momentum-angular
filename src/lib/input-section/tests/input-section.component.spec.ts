import { Component } from '@angular/core';
import {
  async,
  ComponentFixture,
  TestBed,
} from '@angular/core/testing';
import { By } from '@angular/platform-browser';
import { InputSectionModule } from '../../input-section';

@Component({
  template: `
    <md-input-section [className]="className" [position]="position">
      Test
    </md-input-section>
  `,
})

class MDInputSectionComponent {
  className: string;
  position: string;
}

describe('InputSectionComponent', () => {
  let component: MDInputSectionComponent;
  let fixture: ComponentFixture<MDInputSectionComponent>;
  let sectionEl: HTMLElement;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [MDInputSectionComponent],
      imports: [InputSectionModule]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(MDInputSectionComponent);
    component = fixture.componentInstance;
    sectionEl = fixture.debugElement.query(By.css('md-input-section')).nativeElement;
    fixture.detectChanges();
  });

  it('should create and match snapshot', () => {
    expect(component).toBeTruthy();
    expect(fixture).toMatchSnapshot();
  });

  it('should render correct position', () => {
    expect(sectionEl.className.includes('before')).toBe(true);
    fixture.componentInstance.position = 'after';
    fixture.detectChanges();

    expect(sectionEl.className.includes('after')).toBe(true);
  });

  it('should render className', () => {
    expect(sectionEl.className.includes('test-class')).toBeFalsy();
    fixture.componentInstance.className = 'test-class';
    fixture.detectChanges();

    expect(sectionEl.className.includes('test-class')).toBe(true);
  });
});

