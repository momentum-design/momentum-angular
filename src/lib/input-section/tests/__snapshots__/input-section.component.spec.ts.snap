// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputSectionComponent should create and match snapshot 1`] = `
<ng-component>
  <md-input-section
    class="md-input__before"
  >
     Test 
  </md-input-section>
</ng-component>
`;
